runtime/src/lib.rs

```diff
@@ -609,6 +613,67 @@ where
     }
 }
 
+impl OnAggregate<Balance> for Runtime {
+    fn on_aggregate(
+        domain_id: u32,
+        aggregation_id: u64,
+        aggregation: H256,
+        destination: Destination<Balance>,
+    ) -> DispatchResult {
+        /// TODO: define a way to set TIP
+        const RELAYER_TIP: u128 = 10_000;
+
+        /// TODO: Calculate how much this fee is
+        const HYPERBRIDGE_NON_REFUNDABLE_FEE: u128 = 10_000;
+
+        let domain = pallet_aggregate::Domains::<Runtime>::get(domain_id)
+            .ok_or(pallet_aggregate::Error::<Runtime>::UnknownDomainId)?;
+
+        let owner_account = domain
+            .owner
+            .owner()
+            .ok_or(frame_support::error::BadOrigin)?
+            .clone();
+
+        let base_relayer_fee = destination.base_fee.saturating_mul(destination.gas_price);
+        let total_relayer_fee = base_relayer_fee.saturating_add(RELAYER_TIP);
+        let total_fee = total_relayer_fee.saturating_add(HYPERBRIDGE_NON_REFUNDABLE_FEE);
+
+        let held_amount =
+            pallet_aggregate::DomainAmountDispatchFees::<Runtime>::get(domain_id, &owner_account);
+
+        ensure!(
+            held_amount >= total_fee,
+            pallet_aggregate::Error::<Runtime>::InsufficientHeldFundsDispatchFees
+        );
+
+        <Runtime as pallet_aggregate::Config>::Hold::release(
+            &pallet_aggregate::HoldReason::DomainDispatchFee.into(),
+            &owner_account,
+            total_fee,
+            BestEffort,
+        )?;
+
+        pallet_aggregate::DomainAmountDispatchFees::<Runtime>::insert(
+            domain_id,
+            &owner_account,
+            held_amount.saturating_sub(total_fee),
+        );
+
+        pallet_hyperbridge_aggregations::Pallet::<Runtime>::dispatch_aggregation(
+            frame_system::RawOrigin::Signed(owner_account).into(),
+            Params {
```

Comment:
It should contain also the `domain_id` or just rename `aggregation_id` as `id` and use `(u32, u64)` to model the unique id.

Reply:
but why? I don't need the domain id for anything in the actual `dispatch_request` call from ismp, at this point destination and all the params are known

Reply:
ok adding domain_id as param and embedding it in data for dispatch implementation as well